src/ensembl/src/shared/components/upload/Upload.test.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import faker from 'faker';
+import times from 'lodash/times';
+import noop from 'lodash/noop';
+import Upload from './Upload';
+
+const onChange = jest.fn();
+
+const defaultProps = {
+  onChange
+};
+
+const event = { preventDefault: noop, stopPropagation: noop };
+
+const renderUpload = (props: any = defaultProps) =>
+  mount(<Upload {...props} />);
+
+const fileContents = faker.random.words();
+
+const generateFile = () => {
+  return new Blob([fileContents], { type: 'text/plain' });
+};
+
+const files = times(faker.random.number(10), () => generateFile());
+const readAsText = jest.fn();
+const addEventListener = jest.fn((_, evtHandler) => {
+  evtHandler();
+});
+const dummyFileReader = { addEventListener, readAsText, result: fileContents };
+
+describe('Upload', () => {
+  let wrapper: any;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    window.FileReader = jest.fn(() => dummyFileReader) as any;
+    wrapper = renderUpload();
+  });
+
+  describe('using the input', () => {
+    it('always renders an input of type file', () => {
+      expect(wrapper.find('input')).toHaveLength(1);
```

Comment:
description says "of type 'file'"; code isn't checking that

Reply:
I guess I removed it by mistake as I remember adding it 😞